src/backend/utils/misc/queryenvironment.c

```diff
@@ -328,7 +329,8 @@ bool ENRgetSystableScan(Relation rel, Oid indexId, int nkeys, ScanKey key, List
 		reloid != DependRelationId &&
 		reloid != SharedDependRelationId &&
 		reloid != IndexRelationId &&
-		reloid != SequenceRelationId)
+		reloid != SequenceRelationId &&
+		reloid != AttrDefaultRelationId)
```

Comment:
Adding `pg_attrdef` in ENR would look like we regress the following. Have we decided we are okay to take this hit?

```
[clouddesk3] % sqlcmd -S localhost -U jdbc_user -P 1 -e -i ~/test2.sql
CREATE TABLE #temptable(ID INT, ProductName varchar(20) DEFAULT 'product')
INSERT into #temptable (ID) VALUES (1)


(1 rows affected)

BEGIN TRANSACTION
ALTER TABLE #temptable ADD CONSTRAINT df_ProductName DEFAULT 'service' FOR ProductName
INSERT into #temptable (ID) VALUES (2)
select * from #temptable
ROLLBACK


(1 rows affected)
ID          ProductName
----------- --------------------
          1 product
          2 service

(2 rows affected)

INSERT into #temptable (ID) VALUES (3)
select * from #temptable


(1 rows affected)
ID          ProductName
----------- --------------------
          1 product
          3 service. --> should be product because of ROLLBACK

(2 rows affected)
```

Reply:
I think, since this is a byproduct of the ROLLBACK issue we have discussed, I would rather merge `pg_attrdef` now rather than block this entire change until the ROLLBACK issues are fixed. It would be better to space out large changes to ENR rather than merging large changes near the end of the release cycle.

Reply:
Resolving as ENR Rollback is fixed now